More considerations (#39)
lake-wg/psk: draft-ietf-lake-edhoc-psk.md

* Resumption PSKs MUST NOT be used for purposes other than EDHOC session resumption.
* Resumption PSKs MUST be securely stored with the same level of protection as the session keys.
* Parties SHOULD prevent excessive reuse of the same resumption PSK.

> * The external PSK and the resumption PSKs form a key ratchet. If both the external PSK and earlier resumption PSKs are securely destroyed (zeroized), then compromise of a current resumption key does not allow an attacker to reconstruct past PSKs. This property holds regardless of whether ECDHE or a post-quantum key exchange is used.

Reviewer: Re. https://github.com/lake-wg/psk/issues/37#issuecomment-4582267826:

```suggestion
* The external PSK and the resumption PSKs form a key ratchet. If both the external PSK and earlier resumption PSKs are securely destroyed (zeroized), then compromise of a current resumption key does not allow an attacker to reconstruct past PSKs. This property holds regardless of whether ECDHE or a post-quantum key exchange is used.

  Whether the PSK should be deleted once a new resumption key is confirmed (in which case the initiator MUST persist it before TBD and the responder MUST persist it before TBD), or whether the PSK needs to be retained because the resumption keys are volatile state in the peer, needs to be established when the PSK is provisioned.
```

I left the precise points in time as TBD because I'd need to work this out on a flipchart and currently don't have one at hand :-)

Alternatively, something like this could also be in Section 6, which currently only talks about deleting old rPSK.

Reviewer 2: Agree with "needs to be established when the PSK is provisioned.". I think this would be the application template? I and not sure we should use MUST terminology

Reviewer: May be fixed in the application profile, may also be something for which the setup mechanism has an option (eg. when configured from ACE, this can be a field in the token)

> eakage of a PSK does not enable the attacker to calculate past PSKs. 

Reviewer: ```suggestion
  Leakage of a PSK does not enable the attacker to calculate past PSKs. 
```

(editing mistake, or is this line even needed if the reader understands the line above?)

Reviewer 2: We should not have both. Compromise and leakage are just synonyms.

# EDHOC-PSK and OSCORE

> Before sending message_3 the Initiator can derive PRK_out and create an OSCORE-protected request. The request payload MAY convey both an EDHOC message_3 and OSCORE-protected data combined together, as described in {{Section 3 of RFC9668}}. Note that with the use of OSCORE as in {{RFC9668}}, message_4 is not sent, which in EDHOC-PSK results in the Responder not being authenticated in EDHOC. In this case, the Responder is not authenticated until the Initiator has verified a matching OSCORE response. However, only the party with the correct PSK can decrypt the OSCORE request.

Reviewer: I'd suggest doing this more specifically to address #35:

```suggestion
In the previously existing methods, message 4 only gives the Initiator key confirmation. With this method, message 4 is the message which gives the Initiator authentication of the Responder (and also key confirmation).

When EDHOC is used combined with OSCORE as in {{RFC9668}}, message 4 is not sent, but implied by the responder's ability to decrypt the request and create a functional response. Nonetheless, the combined flow can be used: While the responder is not authenticated when message 3 is sent, only an authenticated responder can decrypt message 3.
```

Author: Thanks, I have provided another proposal in #47. (The other changes of this PR are already in the main branch.)
